src/common/test.utils.ts

```diff
@@ -14,6 +17,36 @@ export function appConfig(app: INestApplication) {
   // add cookie parser to read jwt token cookie
   app.use(cookieParser());
   app.useGlobalPipes(new ValidationPipe());
+
+  const configService = app.get(ConfigService);
+  const allowedOrigins = configService.get('ALLOWED_ORIGINS').split(',');
+  for (const origin of allowedOrigins) {
+    if (!isURL(origin)) {
+      throw new Error(`Origin ${origin} is not a valid URL`);
+    }
+  }
+  const restrictCorsOrigins = configService.get<boolean>(
+    'RESTRICT_CORS_ORIGINS'
+  );
+  app.enableCors({
+    credentials: true,
+    origin: restrictCorsOrigins
+      ? function (origin, callback) {
+          if (origin) {
+            if (allowedOrigins.includes(origin)) {
+              callback(undefined, origin);
+            } else {
+              callback(
+                new UnauthorizedException(`Origin ${origin} is not allowed`),
+                false
+              );
+            }
+          } else {
+            callback(undefined, true);
+          }
+        }
+      : true,
+  });
```

Comment:
It occurred to me that this code is duplicating the code in `main.ts`. Maybe we can do it in a separate PR, but is there a way to remove this duplication? Do you think this is worth doing?

Reply:
Yes, this duplication should be removed